### DISCUSSION TITLE
How does the UNO platform render WinUI 3 on Windows?

### DISCUSSION THREAD

**Discussion by ghost** [unoplatform/uno#18611] [Q&A] [answered]
There are basically two ways to render an interface:

1. Call the system's native API to render, e.g. Windows App SDK.
2. Recreate the interface by copying the Design into new code, e.g. imagine a website using HTML/CSS to copy the Design from WinUI3 or GTK-4.

What is the Uno Platform's approach?

**Comment by MartinZikmund** [comment, marked as answer]
@Will-J-Bailey As we have the exact same API surface as WinUI, the WinAppSDK tagret is just a plain WinAppSDK app and apart from some helper classes in Uno.UI.Toolkit, Uno Platform is not involved at all. On the other targets however, Uno Platform provides the same APIs as WinAppSDK does on Windows - hence you are able to take the same "Windows app" codebase and run it everywhere.

So from the two options in your question, 1. is the approach we take
